File: python/Q167_two_sum_ii_input_array_is_sorted.py
#
# @lc app=leetcode id=167 lang=python3
#
# [167] Two Sum II - Input Array Is Sorted
#


# @lc code=start
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        left: int = 0
        right: int = len(numbers) - 1
        result: list[int] = []

        while left < right:
            current: int = numbers[left] + numbers[right]

            if current < target:
                left += 1
            elif current > target:
                right -= 1
            else:
                result = [left + 1, right + 1]
                break

        return result


# @lc code=end

File: rust/src/leetcode/Q167_two_sum_ii_input_array_is_sorted.rs
/*
 * @lc app=leetcode id=167 lang=rust
 *
 * [167] Two Sum II - Input Array Is Sorted
 */

// @lc code=start
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut left: usize = 0;
        let mut right: usize = numbers.len() - 1;

        while left < right {
            let current: i32 = numbers[left] + numbers[right];

            if current < target {
                left += 1;
            } else if current > target {
                right -= 1;
            } else {
                return vec![left as i32 + 1, right as i32 + 1];
            }
        }

        vec![]
    }
}
// @lc code=end
